log_test flakes on Smoke (ext4, false) waiting for 'Disconnected' log

## Summary

The `log_test` step in `tests/workflow/smoke.tests.txt` (test 6/25, runs `tests/lim/testdata/log_test.txt`) intermittently times out on the `Smoke (ext4, false)` matrix entry. The same code passes consistently on the other three Smoke variants. Because `log_test` is early in the smoke sequence, a failure here aborts the run before any of the later 19 tests get to execute, including `ctrl_cert_change` (test 23/25).

## Reproduction

Observed once on PR #1154, [run 25466127075](https://github.com/lf-edge/eden/actions/runs/25466127075). Specific failing job: [Smoke (ext4, false) - 74719820458](https://github.com/lf-edge/eden/actions/runs/25466127075/job/74719820458). A rerun of just the failed jobs has been kicked off; result will be linked here when it lands.

## Trace

```text
=== RUN   TestEdenScripts/log_test
> exec -t 5m bash wait_ssh.sh           # ssh waited 148s before EVE was reachable, then succeeded
> exec sleep 60
> exec -t 27m bash ssh.sh &              # background loop: ssh exit; sleep 10; ...
> test eden.lim.test -test.v -timewait 25m -test.run TestLog \
    -out content 'content:.*Disconnected.*'

    lim_test.go:163: time: 2026-05-06T23:17:49Z out:
        Wait for log of cd40d84b-39d8-4587-8967-d0de9586f083 number=1 timewait=25m0s
    testContext.go:291: WaitForProc terminated by timeout 25m0s
    testContext.go:274: WaitForProcWithErrorCallback terminated by timeout 25m0s

--- FAIL: TestLog (1500.00s)
FAIL: ../lim/testdata/log_test.txt:19: command failure
--- FAIL: TestEdenScripts/log_test (1711.38s)
```

i.e. `ssh.sh` ran for the full 27 min (the test bus killed it at the 25-min `eden.lim.test` timeout), but no log line matching `content:.*Disconnected.*` ever reached the LIM stream from EVE, despite SSH connections being made every 10s.

## Why this looks like a flake, not a regression

Same source tree, same workflow run, same EVE image. Other variants:

| variant | TPM | filesystem | result |
| --- | --- | --- | --- |
| Smoke (ext4, **false**) | off | ext4 | **timeout** |
| Smoke (zfs, false) | off | zfs | pass |
| Smoke (ext4, true) | on | ext4 | pass |
| Smoke (zfs, true) | on | zfs | pass |

The only dimension where this variant is alone is `ext4 + TPM-off`, and the test under inspection has nothing to do with either filesystem or TPM (it only exercises the remote log forwarding path). That makes a real regression unlikely.

## Speculation about the underlying cause

The test depends on three independent things:

1. The two `agent.debug.debug.loglevel=info` / `remote.loglevel=info` config updates being received and applied by EVE before the watcher starts.
2. EVE's sshd actually being reached by the background `ssh.sh` loop and producing the `Disconnected from` message at log level `info`.
3. The log line being forwarded to adam and surfaced through eden's LIM stream within 25 min.

Step 1 is the most timing-sensitive: `wait_ssh.sh` only confirms sshd is up, not that the *config* with the new log levels has been fully applied. The current code follows it with `exec sleep 60`, with a comment specifically calling out that TPM-enabled devices need extra time. A no-TPM run on slower I/O (ext4 on a busy CI runner?) could conceivably need longer too. If the loglevel update has not yet landed when the watcher starts, the SSH disconnects only emit at `debug` level and the watcher misses them.

Step 3 is also worth a look — there are reports of newlogd / log forwarding not always making it to adam under load — but the variant pattern points more at step 1.

## Suggested follow-up

- If the rerun reproduces, capture EVE's `/persist/log/*` and the adam log stream around 23:14–23:43Z so we can tell which of (1)/(2)/(3) is the actual blocker.
- Consider replacing the `exec sleep 60` post-`wait_ssh.sh` with an active wait that confirms the loglevel actually took effect (e.g., `eden controller edge-node get-config | grep loglevel=info`) before kicking off `ssh.sh`. Same fix would help all four matrix entries even though only this one is currently flaking.

## Not blocking PRs #1153 / #1154

Surfacing this so the flake doesn't drag down review of the cert-rotation work. The three other Smoke variants — including `Smoke (ext4, true)`, the same matrix entry that produced the original failure those PRs fix — all reach `ctrl_cert_change` (test 23/25) and pass, so the PR change itself is validated.

--- a/tests/lim/testdata/log_test.txt
+++ b/tests/lim/testdata/log_test.txt
@@ -3,17 +3,24 @@
 # save the current config before changing it
 eden -t 1m controller edge-node get-config --file /tmp/full-config.json
 
+# wait for EVE's sshd before we capture the pre-update lastconfig mtime
+exec -t 5m bash wait_ssh.sh
+
+# snapshot /persist/checkpoint/lastconfig mtime *before* pushing the log-level
+# update so the subsequent active wait can detect when EVE has actually applied
+# the new config
+exec bash capture_checkpoint.sh
+
 # set the remote log level to info, because the expected message is logged with info level
 eden -t 1m controller edge-node update --config agent.debug.debug.loglevel=info
 eden -t 1m controller edge-node update --config agent.debug.debug.remote.loglevel=info
 
-# wait for EVE's sshd to be ready before starting the log watcher
-exec -t 5m bash wait_ssh.sh
-
-# Give EVE time to fetch and apply the updated config.  On TPM-enabled devices the
-# controller config must pass TPM signature verification before EVE applies it; this
-# can take several minutes, far longer than a short fixed sleep.
-exec sleep 60
+# Wait for EVE to fetch and apply the new config (lastconfig mtime advances).
+# This replaces a previous fixed `sleep 60` that intermittently raced with config
+# apply on slower runs (e.g. issue #1156: ext4 + TPM-off Smoke matrix entry).
+# TPM-enabled devices may need extra time for signature verification; a 10m
+# ceiling absorbs that without making the happy path slower.
+exec -t 10m bash wait_config_applied.sh
 
 exec -t 27m bash ssh.sh &
 {{$test}} -out content 'content:.*Disconnected.*'
@@ -39,6 +46,38 @@ until $EDEN eve ssh exit 2>/dev/null; do
   [ "$(date +%s)" -lt "$END" ] || { echo "sshd not ready after 5m"; exit 1; }
   sleep 5
 done
+
+-- capture_checkpoint.sh --
+EDEN={{EdenConfig "eden.root"}}/{{EdenConfig "eden.bin-dist"}}/{{EdenConfig "eden.eden-bin"}}
+# Read the mtime of EVE's applied-device-config file; zedagent rewrites this
+# file each time it accepts and applies a new device config, so its mtime is
+# the same signal evetestkit.WaitForConfigApplied uses.  Watching the file
+# directly (rather than the /persist/checkpoint directory) avoids false
+# positives from unrelated touches on sibling files like lastradioconfig.
+TS=$($EDEN eve ssh 'stat -c %Y /persist/checkpoint/lastconfig' 2>/dev/null | tr -d '\r' | tail -n 1)
+case "$TS" in
+  ''|*[!0-9]*) echo "could not read /persist/checkpoint/lastconfig mtime (got: '$TS')"; exit 1 ;;
+esac
+echo "$TS" > checkpoint_ts
+echo "captured initial lastconfig mtime: $TS"
+
+-- wait_config_applied.sh --
+EDEN={{EdenConfig "eden.root"}}/{{EdenConfig "eden.bin-dist"}}/{{EdenConfig "eden.eden-bin"}}
+INIT=$(cat checkpoint_ts 2>/dev/null)
+case "$INIT" in
+  ''|*[!0-9]*) echo "no initial lastconfig mtime captured"; exit 1 ;;
+esac
+END=$(($(date +%s) + 10*60))
+while [ "$(date +%s)" -lt "$END" ]; do
+  CUR=$($EDEN eve ssh 'stat -c %Y /persist/checkpoint/lastconfig' 2>/dev/null | tr -d '\r' | tail -n 1)
+  case "$CUR" in
+    ''|*[!0-9]*) ;;            # transient SSH/stat failure -- keep polling
+    *) [ "$CUR" != "$INIT" ] && { echo "config applied (lastconfig mtime: $INIT -> $CUR)"; exit 0; } ;;
+  esac
+  sleep 5
+done
+echo "timeout: /persist/checkpoint/lastconfig mtime did not advance from $INIT within 10m"
+exit 1
 
 -- ssh.sh --
 EDEN={{EdenConfig "eden.root"}}/{{EdenConfig "eden.bin-dist"}}/{{EdenConfig "eden.eden-bin"}}

